history_and_modals.md

```diff
@@ -43,13 +44,86 @@ The History API makes handling back clicks for modals rather hard problem. Among
 5. A shared component that would attempt to use the History API in this way
    can easily corrupt a web app's router.
 
+## Precedents in the Web Platform
+
+There are many precedents of a modal-like functionality on the Web Platform today:
+
+- The `dialog`'s `showModal()` mode cancels the dialog on the <kbd>Esc</kbd> key.
+- The `requestFullscreen` mode cancels the full screen on the <kbd>Esc</kbd> key
+  on desktop and on the back button on mobile.
+- The dropdown of a `input[type=date]` is canceled on the <kbd>Esc</kbd> key
+  on desktop and on the back button on mobile.
+
+These features tell us that (a) there's a clear precedent for using the back
+button for cancelling, and (b) this support is platform sensitive: it's expected
+on a mobile platform (especially Android), but not on desktop.
+
 ## Solving history and modals problem
 
 One way to solve this problem is to challenge why the History API should be
 used for modals support at all. The History API integration for a modal is,
 semantically, more similar to <kbd>Esc</kbd> handling than navigation.
 
-A possible solutions, for instance, could require a popup to be built on the
-`<dialog>` element. The browser would provide the Back button integration,
-just as it provides <kbd>Esc</kbd> support today. The dialog API would also
-need to support "save data" prompt.
+Some of the possible solutions could include:
+
+1. An extended `<dialog>` API or a new platform primitive.
+2. A back button event propagated via the <kbd>Esc</kbd> key in some cases.
+
```

Comment:
Do you think a new event type, like "back button key" would be another solution? Or is there some reason to prefer the Esc key event doing double duty

Reply:
Hmm. Yeah, that's probably a good idea. Just got carried away with, what if all existing dialogs started to magically work...

Reply:
Added a new section for the new event type.